Core/src/TrackFinding/MeasurementSelector.cpp

```diff
@@ -67,4 +91,41 @@ double MeasurementSelector::calculateChi2(
       });
 }
 
+MeasurementSelector::Cuts MeasurementSelector::getCutsByTheta(
+    const MeasurementSelectorCuts& config, double theta) {
+  std::size_t bin = 0;
+
+  if (!config.etaBins.empty()) {
+    const double eta = std::atanh(std::cos(theta));
+    const double etaAbs = std::abs(eta);
+    for (; bin < config.etaBins.size(); bin++) {
+      if (config.etaBins[bin] >= etaAbs) {
+        break;
+      }
+    }
+  }
+
+  const double chi2CutOffMeasurement = config.chi2CutOff.at(bin);
+  const double chi2CutOffOutlier = config.chi2CutOffOutlier.empty()
+                                       ? std::numeric_limits<double>::infinity()
+                                       : config.chi2CutOffOutlier.at(bin);
+  const std::size_t numMeasurementsCutOff =
+      config.numMeasurementsCutOff.at(bin);
```

Comment:
For each vector, it used to be possible that `bin >= cut.size()`, because we did not have to specify the cut for all eta bins. The convention we had before (and is used in Athena) was that it used the the last-bin cut for all subsequent bins. This is convenient to allow to specify a single element for cuts that are not eta-dependent, even if we have eta bins for the other cuts.

I see that now `validateCuts` checks that all bins are specified, so it will not crash here. Rather, I guess, `validateCuts` will fail for configurations that previously worked.

Isn't it easier to limit the `bin` index for each `vector`. This could be done with a simple lambda. Unfortunately, a lambda can't handle the empty vector case, because of the different no-cut values for `numMeasurementsCutOff`.

Reply:
Right this is what I had in a previous commit but @goetzgaycken and I agreed on simplifying it. I will try to make this non breaking again

Reply:
```
auto lim = [](const auto& vec, std::size_t bin) { return bin<vec.size() ? vec[bin] : vec.back(); };
```

Reply:
Sorry I missed @goetzgaycken's comment before. I could have pushed back a bit at the time. Quite apart from the incompatibility with previous behaviour, I think it is useful to have this flexibility in the configuration.

Reply:
Not that I am saying you should necessarily use this, but I was pleased to find the general limit lambda:
```
    auto lim = [](const auto& vec, std::size_t bin) {
        return vec.empty()
                   ? std::numeric_limits<std::remove_reference_t<decltype(vec[0])>>::max()
               : bin < vec.size() ? vec[bin]
                                  : vec.back();
    };
```
(returns max `double`, rather than `inf`, but that's OK I think.)